### PR TITLE
AUTH-73 feat(rtp): example RTP for testing

### PR DESCRIPTION
**Pull request by y3rsh**
## Overview
<https://opentrons.atlassian.net/browse/AUTH-73>
<https://opentrons.atlassian.net/browse/RDEVOPS-71>

~~Examples only.~~
~~This PR will not integrate these examples into the analyses snapshot.~~
### I lied.

### CODE REVIEW

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[bda5ad87f9][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_description].json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      "detail": "ParameterValueError [line 84]: Parameter value has type <class 'int'> must match type <class 'str'>.",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "opentrons.protocols.parameters.types.ParameterValueError: Parameter value has type <class 'int'> must match type <class 'str'>.",
+          "errorCode": "4000",
+          "errorInfo": {
+            "args": "(\"Parameter value has type <class 'int'> must match type <class 'str'>.\",)",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[bda5ad87f9][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_description].json]: Good error.

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[da89c9580a][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_choice_display_name].json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      "detail": "TypeError [line 51]: object of type 'float' has no len()",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "TypeError: object of type 'float' has no len()",
+          "errorCode": "4000",
+          "errorInfo": {
+            "args": "(\"object of type 'float' has no len()\",)",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[da89c9580a][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_choice_display_name].json]: mid error

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[b4d4b33215][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_minimum].json [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      "detail": "ParameterDefinitionError [line 95]: Only parameters of type float or int can have a minimum and maximum",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "opentrons.protocols.parameters.types.ParameterDefinitionError: Only parameters of type float or int can have a minimum and maximum",
+          "errorCode": "4000",
+          "errorInfo": {
+            "args": "('Only parameters of type float or int can have a minimum and maximum',)",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[b4d4b33215][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_minimum].json]: I think my param is messed up or this is an issue.

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[b3359925b7][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_maximum].json [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      "detail": "ParameterDefinitionError [line 104]: Only parameters of type float or int can have a minimum and maximum",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "opentrons.protocols.parameters.types.ParameterDefinitionError: Only parameters of type float or int can have a minimum and maximum",
+          "errorCode": "4000",
+          "errorInfo": {
+            "args": "('Only parameters of type float or int can have a minimum and maximum',)",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[b3359925b7][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_maximum].json]: check the param.

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[9cb13be07e][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_unit].json [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      "detail": "TypeError [line 113]: object of type 'int' has no len()",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "TypeError: object of type 'int' has no len()",
+          "errorCode": "4000",
+          "errorInfo": {
+            "args": "(\"object of type 'int' has no len()\",)",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[9cb13be07e][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_unit].json]: mid error

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[89be288557][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_default].json [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      "detail": "ParameterValueError [line 73]: Parameter value has type <class 'int'> must match type <class 'str'>.",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "opentrons.protocols.parameters.types.ParameterValueError: Parameter value has type <class 'int'> must match type <class 'str'>.",
+          "errorCode": "4000",
+          "errorInfo": {
+            "args": "(\"Parameter value has type <class 'int'> must match type <class 'str'>.\",)",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[89be288557][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_default].json]: Good error.

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[84ca834292][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_choice_value].json [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      "detail": "ParameterDefinitionError [line 62]: All choices provided must match type <class 'type'>",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "opentrons.protocols.parameters.types.ParameterDefinitionError: All choices provided must match type <class 'type'>",
+          "errorCode": "4000",
+          "errorInfo": {
+            "args": "(\"All choices provided must match type <class 'type'>\",)",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[84ca834292][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_choice_value].json]: Good error.

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[70e0807365][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_display_name].json [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      "detail": "TypeError [line 30]: object of type 'int' has no len()",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "TypeError: object of type 'int' has no len()",
+          "errorCode": "4000",
+          "errorInfo": {
+            "args": "(\"object of type 'int' has no len()\",)",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[70e0807365][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_display_name].json]: mid error

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[5959cb6f9c][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_variable_name].json [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+      "detail": "AttributeError [line 39]: 'dict' object has no attribute 'isidentifier'",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "AttributeError: 'dict' object has no attribute 'isidentifier'",
+          "errorCode": "4000",
+          "errorInfo": {
+            "args": "(\"'dict' object has no attribute 'isidentifier'\",)",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[5959cb6f9c][Flex_None_None_2_18_BadTypesInRTP_Override_wrong_type_in_variable_name].json]: less than mid error

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[27c2119f32][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_variable_name].json [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+  },
+  "errors": [
+    {
+      "detail": "AttributeError [line 39]: 'dict' object has no attribute 'isidentifier'",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "AttributeError: 'dict' object has no attribute 'isidentifier'",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[27c2119f32][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_variable_name].json:26]: https://opentrons.atlassian.net/browse/AUTH-303

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[7a9449b64c][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_choice_value].json [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+  },
+  "errors": [
+    {
+      "detail": "ParameterDefinitionError [line 62]: All choices provided must match type <class 'type'>",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "opentrons.protocols.parameters.types.ParameterDefinitionError: All choices provided must match type <class 'type'>",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[7a9449b64c][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_choice_value].json:26]: https://opentrons.atlassian.net/browse/AUTH-304

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[8e15076a97][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_choice_display_name].json [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+  },
+  "errors": [
+    {
+      "detail": "TypeError [line 51]: object of type 'float' has no len()",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "TypeError: object of type 'float' has no len()",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[8e15076a97][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_choice_display_name].json:26]: https://opentrons.atlassian.net/browse/AUTH-305

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[d13f3b33af][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_maximum].json [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+  "config": {
+    "apiVersion": [
+      2,
+      18
+    ],
+    "protocolType": "python"
+  },
+  "errors": [
+    {
+      "detail": "ParameterDefinitionError [line 104]: Only parameters of type float or int can have a minimum and maximum",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[d13f3b33af][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_maximum].json:20]: https://opentrons.atlassian.net/browse/AUTH-306

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[d6a37191cf][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_minimum].json [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+  "config": {
+    "apiVersion": [
+      2,
+      18
+    ],
+    "protocolType": "python"
+  },
+  "errors": [
+    {
+      "detail": "ParameterDefinitionError [line 95]: Only parameters of type float or int can have a minimum and maximum",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[d6a37191cf][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_minimum].json:20]: https://opentrons.atlassian.net/browse/AUTH-306

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[da8add28b8][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_display_name].json [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+  "config": {
+    "apiVersion": [
+      2,
+      18
+    ],
+    "protocolType": "python"
+  },
+  "errors": [
+    {
+      "detail": "TypeError [line 30]: object of type 'int' has no len()",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[da8add28b8][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_display_name].json:20]: https://opentrons.atlassian.net/browse/AUTH-307

File: app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[e3ad5463a4][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_unit].json [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+  },
+  "errors": [
+    {
+      "detail": "TypeError [line 113]: object of type 'int' has no len()",
+      "errorCode": "4000",
+      "errorInfo": {},
+      "errorType": "ExceptionInProtocolError",
+      "wrappedErrors": [
+        {
+          "detail": "TypeError: object of type 'int' has no len()",
```
Comment (y3rsh) [inline app-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[e3ad5463a4][v2_18_NO_PIPETTES_Overrides_BadTypesInRTP_Override_wrong_type_in_unit].json:26]: https://opentrons.atlassian.net/browse/AUTH-307

**Comment by DerekMaggio** [comment]
I did notice an issue when I was changing all the file names/paths. Because we have removed API version and robot type from the file name, we are starting to have duplicate file names. In the flat list we have of all the file names, this is going to become an issue. 

Instead of the file names, should we make that list the file path from the `files` directory?

**Comment by y3rsh** [comment]
> I did notice an issue when I was changing all the file names/paths. Because we have removed API version and robot type from the file name, we are starting to have duplicate file names. In the flat list we have of all the file names, this is going to become an issue.
> 
> Instead of the file names, should we make that list the file path from the `files` directory?

Everything back into filename but in a better organizing pattern.

File: app-testing/files/protocols/README.md [hunk 17]
```diff
@@ -1,34 +1,33 @@
...
-2.16_None_None_AnalysisError_TrashBinInStagingAreaCol4.py
-2.16_P1000_96_GRIPPER_HS_TM_TC_MB_DeckConfiguration1_NoFixtures.py
+OT2_2.18_None_None_Overrides_BadTypesInRTP.py
+OT2_2.16_None_None_AnalysisError_TrashBinInStagingAreaCol4.py
+OT2_2.16_P1000_96_GRIPPER_HS_TM_TC_MB_DeckConfiguration1_NoFixtures.py
 
 #### .json
 
-5_2_6_P300M_P20S_MM_TM_TC1_PD40.json
+OT2_5_2_6_P300M_P20S_MM_TM_TC1_PD40.json
```
Comment (DerekMaggio) [inline app-testing/files/protocols/README.md]:
```suggestion
OT2_S_v2_18_None_None_Overrides_BadTypesInRTP.py
OT2_X_v2_16_None_None_TrashBinInStagingAreaCol4.py
Flex_S_v2_15_P1000_96_GRIP_HS_TM_QuickZymoMagbeadRNAExtraction,

#### .json

Flex_X_v8_P1000_96_HS_GRIP_TC_TM_GripperCollisionWithTips.json
```

**Review by DerekMaggio — COMMENTED** [review]
This newest commit looks good for renaming the files.

I think updating the README to describe how ProtocolWithOverrides is intended to be used when generating protocols would be helpful as well

File: app-testing/files/protocols/README.md (hunk 17 above)
Comment (y3rsh) [reply app-testing/files/protocols/README.md]: yup changed this updating readme

**Review by DerekMaggio — APPROVED** [review]
Looks good to me

**Pull request merged**